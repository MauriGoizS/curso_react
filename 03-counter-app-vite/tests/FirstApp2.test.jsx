const { render, screen } = require("@testing-library/react")
const { FirstApp } = require("../src/FirstApp")

describe('Tests in <FirstApp />', () => {

    const title = 'Hi, I´m Goku';
    const subTitle = 'I´m subtitle';

    test('Must match the snapshot ', () => {
    
        const { container } =  render( <FirstApp title={ title }/>);
        expect( container ).toMatchSnapshot();

    })
    
    test('It should display the message "Hello, I am Goku"', () => {
    
        render( <FirstApp title={ title }/>);
        expect( screen.getByText( title ) ).toBeTruthy();

    })
    
    test('Must display the title in a h1', () => {
    
        render( <FirstApp title={ title }/>);
        expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain( title );

    })

    test('Must match the subtitle send via props', () => {
    
        render(
            <FirstApp 
                title={ title }
                subTitle= { subTitle }
            />
        );

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    });
    
    

});
