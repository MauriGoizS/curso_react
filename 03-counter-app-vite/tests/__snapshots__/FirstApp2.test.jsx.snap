// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Must match the snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hi, I´m Goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Mauricio Goiz
  </p>
</div>
`;
